realm/rfc/0001-backlink-query-syntax.md

```diff
@@ -0,0 +1,232 @@
+- Feature Name: Inverse relationship syntax
+- Start Date: 2016-05-30
+- RFC PR: 
+- Version: 3
+
+# Summary
+
+It should be possible to define and query backlinks in both `Realm` and 
+`DynamicRealm`. As a `DynamicRealm` can work without _any_ Realm model 
+classes, it should be possible to query backlinks without them being defined there.
+
+# Motivation
+
+Backlinks are special in Realm. They are not defined explicitly in the schema,
+but are automatically created when defining Links and LinkLists (forward links).
+
+This creates some conflicts with the current query system as it rely on all 
+fields being defined by the schema. This is called *named backlinks*.
+
+This proposal introduces a new syntax for queries so it is possible to query
+across a backlink even though it hasn't been defined by a model class. This is
+called *unnamed backlinks*.
+
+Any design should ideally solve all these constraints:
+
+- Should support a list of backlinks.
+
+- Should support a single backlink, enforcing the constraint that only 1 object 
+  can point to it. This is not supported by Core, but it is a natural extension.
+
+- Should support backlinks from multiple fields.
+
+- Should be usable from the Dynamic API.
+
+- Should either support polymorphic objects or not prevent them from being 
+  implemented.
+
+- Ideally it should work for both a pure text-based variant as well as the 
+  current mixed style found in Realm Java.
+  
+
+# Detailed design
+
+**Example model classes:**
+
+```
+public class Person extends RealmObject {
+	private Dog favoriteDog;
+	private RealmList<Dog> dogs;
+}
+
+public class Dog extends RealmObject {
+	private String name
+
+	// Named backlinks below
+	@Backlink({"favoriteDog"})
```

Comment:
Is `Backlink` the best term to use in the API for this? On the Cocoa side of the world we've completely avoided using this term in our API and documentation since its meaning is not obvious to users without further explanation. I feel like we should make an attempt to keep the terminology used for a given concept consistent between bindings.